airflow-core/src/airflow/models/connection.py

```diff
@@ -529,6 +529,18 @@ def to_dict(self, *, prune_empty: bool = False, validate: bool = True) -> dict[s
 
         :meta private:
         """
+        if hasattr(sys.modules.get("airflow.sdk.execution_time.task_runner"), "SUPERVISOR_COMMS"):
```

Comment:
Temporary compat layer for 3.1

Reply:
Oh fair point, we won't need it for instance methods. Only classmethods will do the trick

Reply:
Nope, handled in https://github.com/apache/airflow/pull/53870/commits/c9b558bac1876024138fb16c27b950a535000dd7